var/spack/repos/builtin/packages/py-merlin/package.py

```diff
@@ -23,14 +24,18 @@ class PyMerlin(PythonPackage):
     depends_on("python@3.6:", type=("build", "run"))
     depends_on("py-setuptools", type=("build", "run"))
     depends_on("py-cached-property", type=("build", "run"))
-    depends_on("py-celery@5.0.0+redis+sqlalchemy", when="@1.7.5:", type=("build", "run"))
+    depends_on("py-celery@5.0.3:+redis+sqlalchemy", when="@1.8.0:", type=("build", "run"))
+    depends_on("py-celery@4.4.5:+redis+sqlalchemy", when="@1.7.5:", type=("build", "run"))
     depends_on("py-celery@4.3.0:4+redis+sqlalchemy", when="@:1.7.4", type=("build", "run"))
```

Comment:
According to [release.txt](https://github.com/LLNL/merlin/blob/1.10.3/requirements/release.txt) the releases from where the versions change are different:
```suggestion
    depends_on("py-celery@5.0.3:+redis+sqlalchemy", when="@1.7.6:", type=("build", "run"))
    depends_on("py-celery@5.0.0:+redis+sqlalchemy", when="@1.7.5", type=("build", "run"))
    depends_on("py-celery@4.4.5:+redis+sqlalchemy", when="@1.6.2:", type=("build", "run"))
    depends_on("py-celery@4.3.0:+redis+sqlalchemy", when="@:1.5.3", type=("build", "run"))
```

Reply:
Hm, i looked at git blame from the main branch, I guess they backport changes to release.txt

Reply:
The last bound doesn't make sense:

```python
depends_on("py-celery@4.3.0:+redis+sqlalchemy"
```

either drop when or set an upperbound in both the dep and when condition :thinking: 

I guess the upperbound `@:4` is there for a reason

Reply:
Incorporated your changes, but kept the upperbound, since it's likely that old releases forgot to set an upperbound on the major version.

Reply:
You are right there was a upper bound missing in the when condition. Usually we don't set version restrictions on the dependencies for that because it makes understanding the reasoning behind it for future reviews harder.